samples/bluetooth/peripheral_hr/src/main.c

```diff
@@ -126,24 +117,126 @@ static void hrs_notify(void)
 	}
 }
 
+#if defined(CONFIG_GPIO)
+/* The devicetree node identifier for the "led0" alias. */
+#define LED0_NODE DT_ALIAS(led0)
+
+#if DT_NODE_HAS_STATUS(LED0_NODE, okay)
+#include <zephyr/drivers/gpio.h>
+#define HAS_LED     1
+static const struct gpio_dt_spec led = GPIO_DT_SPEC_GET(LED0_NODE, gpios);
+#define BLINK_ONOFF K_MSEC(500)
+
+static struct k_work_delayable blink_work;
+static bool                  led_is_on;
+
+static void blink_timeout(struct k_work *work)
+{
+	led_is_on = !led_is_on;
+	gpio_pin_set(led.port, led.pin, (int)led_is_on);
+
+	k_work_schedule(&blink_work, BLINK_ONOFF);
+}
+#endif
+#endif /* CONFIG_GPIO */
+
 int main(void)
```

Comment:
`main` is now 150 lines of code. Suggest to move some of the code blocks into functions, so you have functions like

```
configure_gpio
start_advertising
start_led
stop_led
```

to make the flow of the `main` easier to follow.